File: src/main/java/com/aquasecurity/plugins/trivy/model/Misconfiguration.java
package com.aquasecurity.plugins.trivy.model;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

public class Misconfiguration {
    @JsonProperty("Type")
    public String type;
    @JsonProperty("ID")
    public String id;
    @JsonProperty("Title")
    public String title;
    @JsonProperty("Description")
    public String description;
    @JsonProperty("Message")
    public String message;
    @JsonProperty("Namespace")
    public String namespace;
    @JsonProperty("Query")
    public String query;
    @JsonProperty("Resolution")
    public String resolution;
    @JsonProperty("Severity")
    public String severity;
    @JsonProperty("PrimaryURL")
    public String primaryURL;
    @JsonProperty("References")
    public List<String> references;
    @JsonProperty("Status")
    public String status;
    @JsonProperty("Layer")
    public Layer layer;
    @JsonProperty("IacMetadata")
    public IacMetadata iacMetadata;
}

File: src/main/kotlin/com/aquasecurity/plugins/trivy/model/report/Misconfiguration.kt
package com.aquasecurity.plugins.trivy.model.report

import com.fasterxml.jackson.annotation.JsonProperty

class Misconfiguration {
  @JsonProperty("Type") var type: String? = null

  @JsonProperty("ID") var id: String? = null

  @JsonProperty("Title") var title: String? = null

  @JsonProperty("Description") var description: String? = null

  @JsonProperty("Message") var message: String? = null

  @JsonProperty("Namespace") var namespace: String? = null

  @JsonProperty("Query") var query: String? = null

  @JsonProperty("Resolution") var resolution: String? = null

  @JsonProperty("Severity") var severity: String? = null

  @JsonProperty("PrimaryURL") var primaryURL: String? = null

  @JsonProperty("References") var references: List<String>? = null

  @JsonProperty("Status") var status: String? = null

  @JsonProperty("Layer") var layer: Layer? = null

  @JsonProperty("IacMetadata") var iacMetadata: IacMetadata? = null

  @JsonProperty("CauseMetadata") var causeMetadata: IacMetadata? = null
}
